arxiv-base/Dockerfile

```diff
@@ -0,0 +1,30 @@
+# arxiv/flask base docker file
+
+FROM centos:centos7
+
+RUN yum -y update && yum -y install epel-release \
+&& yum -y install https://centos7.iuscommunity.org/ius-release.rpm \  
+&& echo $'#!/bin/bash\n\
+PKGS_TO_INSTALL=$(cat <<-END\n\
+  ca-certificates\n\
+  wget\n\
+  gcc\n\
+  gcc-c++ \n\
+  python36u\n\
+  python36u-devel\n\
+  git\n\
+END\n\
+)\n\
+for pkg in ${PKGS_TO_INSTALL}; do\n\
+  # Stop executing if at least one package is not available:\n\
```

Comment:
This might more accurately say stop executing chained command if the last package is not available.  Add set -e
./for_command_exit.sh
running true
running true
running false
command failed: false
running true
continued
keep going good

The continued output is bad.

#!/bin/bash

# full commands are difficult to get space interpretation correct, deferring
# COMMANDS=$(cat <<-END
# echo
# echo hi
# echo foo
# END
# )
#
# #exit 1


COMMANDS=$(cat <<-END
true
true
false
true
END
)

#export IFS='\n' # unnecessary
for cmd in ${COMMANDS}; do
  echo "running ${cmd}"
  # Attempt to Stop executing after loop if at least one command fails
  ${cmd} || ( echo "command failed: ${cmd}"; exit 1 )
done \
&& echo "continued"

echo "keep going good"
exit 0

Reply:
Revision: after testing, set -e alone is not enough, setting and checking a flag works

```cat set_flag_for_command_exit.sh
#!/bin/bash

COMMANDS=$(cat <<-END
true
true
false
true
END
)

export good=0
for cmd in ${COMMANDS}; do
  echo "running ${cmd}"
  # Stop executing after loop if at least one command fails
  ${cmd} || {
    echo "command failed: ${cmd}"
    good=1
  }
  echo "$good"
  [ $good -eq 0 ]
done \
&& echo "continued"

echo "keep going good"
exit 0
```
```running true
0
running true
0
running false
command failed: false
1
running true
1
keep going good```

Reply:
the exit 1 does not break the loop because it is inside a subshell () inside the for loop rather than  a code block as above {}
https://stackoverflow.com/questions/14059342/how-to-get-the-exit-status-a-loop-in-bash

Reply:
@rcplane In your output we see "keep going good" - so we have to be careful to make the  script end of the for loop, else it will not stop docker build. I may write a function as suggested in the SO post.

Reply:
Fixed using {} as discussed